### ISSUE TITLE
confuced of the twice the performance of these method like MPE.

### ISSUE THREAD

**Issue by momo-xiaoyi** [samjia2000/HSP#2]
Compared with the original mpe and FCP peformance, the author's reproduce experiment is twice of the original performance. However, my reproduced version of these algorithm can't improve that much, which makes me confused.

What caused the algorithm to improve so much? Is anyone as confused as I am?

**Comment by momo-xiaoyi** [comment]
I think i find the problem! In the 214 line of file [(https://github.com/samjia2000/HSP/tree/main/hsp/runner/shared)/overcooked_runner.py ](https://github.com/samjia2000/HSP/blob/main/hsp/runner/shared/overcooked_runner.py) 
` eval_env_infos['eval_average_episode_rewards'] = np.sum(eval_average_episode_rewards, axis=0)` 

It calclulate the sum of the "eval_average_episode_rewards", however "eval_average_episode_rewards" is already the sum of sparse reward by each agent.

This makes a double perfoemance. a mistake in code.

**Comment by samjia2000** [comment]
Regarding the performance gap, in general, we carefully tune the performance of MEP and FCP. For example, when training the adaptive policy, the original MEP implementation only trains against one policy from the policy pool in one PPO iteration. However, in our implementation, the adaptive policy plays with a number of sampled policies from the policy pool in one PPO iteration. To be more explicit, in every PPO iteration, the environment threads independently sample policies from the pool. This significantly reduces the variance during training. Also, hyperparameters such as batch size could influence the PPO performance a lot.
